studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx

```diff
@@ -0,0 +1,211 @@
+import { useMemo, useState } from "react";
+
+const inputTheme = EditorView.theme({
+  "&": {
+    background: "transparent !important",
+    // HACK - Using tailwind `text-sm` values
+    fontSize: "0.875rem",
+    lineHeight: "1.25rem",
+  },
+  // Hide the gutters
+  ".cm-gutters": {
+    borderColor: "transparent",
+    backgroundColor: "transparent",
+  },
+  ".cm-cursor": {
+    borderLeftColor: "white",
+  },
+});
+
+// Configuration to disable all basic setup features on code mirror instance, so it behaves more like a text input
+const CODE_MIRROR_BASIC_SETUP_DISABLE_ALL = {
+  lineNumbers: false,
+  foldGutter: false,
+  dropCursor: false,
+  allowMultipleSelections: false,
+  indentOnInput: false,
+  bracketMatching: false,
+  closeBrackets: false,
+  autocompletion: false,
+  rectangularSelection: false,
+  highlightActiveLine: false,
+  highlightSelectionMatches: false,
+  closeBracketsKeymap: false,
+  defaultKeymap: false,
+  searchKeymap: false,
+  historyKeymap: false,
+  foldKeymap: false,
+  completionKeymap: false,
+  lintKeymap: false,
+};
+
+// An extension that does nothing to make conditional extensions easier,
+// e.g., `isFocused ? noopExtension : inputTrucateExtension`
+const noopExtension = EditorView.theme({});
+
+// Apply styles for the editor to make it look like one of our inputs
+const inputBaseStylesExtension = EditorView.theme({
+  ".cm-placeholder": {
+    color: "hsl(var(--muted-foreground))",
+    // HACK - Using tailwind `text-sm` values
+    fontSize: "0.875rem",
+    lineHeight: "1.25rem",
+  },
+  ".cm-scroller": {
+    overflow: "hidden !important",
+    maxWidth: "100% !important",
+  },
+});
+
+// Extension to truncate text that overflows with an ellipsis
+// We only want to use this when the editor is *not* focused
+const inputTrucateExtension = EditorView.theme({
+  ".cm-content": {
+    whiteSpace: "nowrap",
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+    width: "100%",
+  },
+  ".cm-line": {
+    textOverflow: "ellipsis",
+    overflow: "hidden",
+    width: "100%",
+  },
+});
+
+// Extension for special styling in readonly mode, which makes the editor's cursor transparent,
+// but uses a pointer cursor when the user hovers over the input,
+// to indicate that you can click on it to expand it (in case the path parameter is truncated).
+// As of writing, this is only used for the path parameter *key* input.
+const readonlyExtension = EditorView.theme({
+  ".cm-cursor": {
+    borderLeftColor: "transparent !important",
+  },
+  ".cm-content": {
+    cursor: "pointer !important",
+  },
+});
+
+// Hide the gutter on the code mirror instance
+const hiddenGutterExtension = gutter({ class: "hidden border-none" });
+
+type CodeMirrorEditorProps = {
+  readOnly?: boolean;
+  width?: string;
+  value?: string;
+  onChange: (value?: string) => void;
+  placeholder?: string;
+  onSubmit?: () => void;
+};
+
+const preventNewlineInFirefox = keymap.of([
+  {
+    key: "Enter",
+    run: (_view) => {
```

Comment:
Curious, can this parameter be removed?

Reply:
indeed it can - was toying with it to see if i had additional info on (e.g.) whether or not "mod" was pressed as well